src/cpp/rtps/builtin/discovery/participant/PDP.cpp

```diff
@@ -429,6 +429,24 @@ bool PDP::enable()
     return builtin_endpoints_->enable_pdp_readers(mp_RTPSParticipant);
 }
 
+void PDP::disable()
+{
+    // Extract all the participant proxies excluding first one (ourselves)
+    std::vector<ParticipantProxyData*> participants;
+    {
+        std::lock_guard<std::recursive_mutex> guardPDP(*this->mp_mutex);
+        participants.insert(participants.end(), participant_proxies_.begin() + 1, participant_proxies_.end());
+        participant_proxies_.erase(participant_proxies_.begin() + 1, participant_proxies_.end());
+    }
+
+    // Unmatch all remote participants
+    for (ParticipantProxyData* pdata : participants)
```

Comment:
The code in `remove_remote_participant` takes the PDP mutex only to search and erase the proxy from `participant_proxies_`.

I mimicked the behavior here, so the mutex only protects the manipulation on `participant_proxies_`, and `remote_participant_removed()` with the mutex unlocked.

This needs to be this way to avoid ABBA lock issues.